### PR TITLE
Un coerce independent vars

### PR DESCRIPTION
**Pull request by newville**
#### Description

This changes the coercion rules for the results of `minimizer` and data and independent variables for `Model`.

It addresses some of the discussion in #850.
 
###### Type of Changes

- [x] Bug fix
- [ ] New feature
- [x] Refactoring / maintenance
- [ ] Documentation / examples

###### Tested on

Python: 3.9.15 (main, Nov 24 2022, 08:29:02)
[Clang 14.0.6 ]

lmfit: 1.1.0.post9+gd2ba8e9.d20230313, scipy: 1.9.3, numpy: 1.23.5, asteval: 0.9.29, uncertainties: 3.1.7

###### Verification 
Have you

- [ ] included docstrings that follow PEP 257?

- [x] referenced existing Issue and/or provided relevant link to mailing list?

- [x] verified that existing tests pass locally?

- [x] verified that the documentation builds locally?

- [ ] squashed/minimized your commits and written descriptive commit messages?

- [x] added or updated existing tests to cover the changes?
- [ ] updated the documentation and/or added an entry to the release notes (doc/whatsnew.rst)?
- [ ] added an example?

### CODE REVIEW

File: lmfit/model.py [hunk 1]
```diff
@@ -1027,22 +1027,23 @@ def fit(self, data, params=None, weights=None, method='leastsq',
...
+            """
+            if isinstance(x, (list, tuple, Series)):
+                if np.isrealobj(x):
+                    return np.asfarray(x)
+                elif np.iscomplexobj(x):
+                    return np.asarray(x, dtype='complex128')
+            return x
+
+        # coerce array-like (list, tuples, Series) data and
+        # independent variable(s) to float64/complex128
```
Comment (reneeotten) [inline lmfit/model.py:1043]: is this comment completely accurate? It seems to me from the code that independent variables will only be coerced if the are lists/tuples/pandas.Series. The way I read the comment is that data will be coerced if belonging to the categories listed and that independent variable(s) will always be coerced.

**Review by reneeotten — COMMENTED** [review]
@newville thanks - perhaps re-read the comment in the code, to me it's a bit unclear/inaccurate but that could just be me ;)

File: lmfit/model.py (hunk 1 above)
Comment (newville) [reply lmfit/model.py:1043]: will try to clarify the comment.

**Comment by newville** [comment]
@reneeotten Thanks - updated comment!

**Pull request merged**